.github/workflows/rust.yml

```diff
@@ -0,0 +1,78 @@
+name: Rust
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+  merge_group:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_protected == 'true' && github.sha || github.ref }}-{{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+
+  complete:
+    if: always()
+    needs: [fmt, build-and-test]
+    runs-on: ubuntu-latest
+    steps:
+    - if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
+      run: exit 1
+
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: rustup update
+    - run: cargo fmt --all --check
+
+  build-and-test:
+    strategy:
+      matrix:
+        rust: [msrv, latest]
+        sys:
+        - os: ubuntu-latest
+          target: x86_64-unknown-linux-gnu
+        - os: macos-latest
+          target: x86_64-apple-darwin
+        - os: macos-latest
+          target: aarch64-apple-darwin
+        - os: windows-latest
+          target: x86_64-pc-windows-msvc
+    runs-on: ${{ matrix.sys.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: stellar/actions/rust-cache@main
+    - name: Use the minimum supported Rust version
+      if: matrix.rust == 'msrv'
+      run: |
+        msrv="$(cargo metadata --format-version 1 --no-deps | jq -r '.packages | map(.rust_version) | map(values) | min')"
+        rustup override set $msrv
+        rustup component add clippy --toolchain $msrv
+    - name: Error on warnings and clippy checks
+      # Only error on warnings and checks for the msrv, because new versions of
+      # Rust will frequently add new warnings and checks.
+      if: matrix.rust == 'msrv'
+      run: echo RUSTFLAGS='-Dwarnings -Dclippy::all -Dclippy::pedantic' >> $GITHUB_ENV
+    - run: rustup update
+    - run: rustup target add ${{ matrix.sys.target }}
+    - run: cargo version
+    - run: cargo clippy --target ${{ matrix.sys.target }} --all-targets
+    - run: cargo build --target ${{ matrix.sys.target }}
+    # Tests can only run on native targets, not cross-compilation targets like aarch64-apple-darwin
+    - name: Run tests
+      if: matrix.sys.target != 'aarch64-apple-darwin'
+      run: cargo test --target ${{ matrix.sys.target }}
```

Comment:
If you'd like to run the tests on all targets, change the x86 macos build to run on the `macos-13` runner. It's an Intel chip runner.